File: spark-2.4/SparkAvro/src/main/java/com/ranga/spark/avro/SparkAvroReadExample.java
package com.ranga.spark.avro;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

/* rangareddy.avula created on 11/06/20 */

public class SparkAvroReadExample {
    public static void main(String[] args) {

        SparkSession spark = SparkSession.builder().master("local[1]")
                .appName("Spark Avro Read Examples")
                .getOrCreate();

        Dataset<Row> usersDF = spark.read().format("avro").load("./src/main/resources/users.avro");
        usersDF.printSchema();
        usersDF.foreach(row -> {System.out.println(row);});
        spark.close();
    }
}

File: spark-2.3/SparkAvro/src/main/scala/com/ranga/spark/avro/SparkAvroReadExample.scala
package scala.com.ranga.spark.avro

import org.apache.spark.SparkConf
import org.apache.spark.sql.SparkSession

/* rangareddy.avula created on 11/06/20 */

object SparkAvroReadExample {
    def main(args: Array[String]): Unit = {
        val conf = new SparkConf().setIfMissing("spark.master", "local[*]").setAppName("Spark Avro Read Examples")
        val spark = SparkSession.builder().config(conf).getOrCreate();

        val usersDF = spark.read.format("com.databricks.spark.avro").load("./src/main/resources/users.avro");
        usersDF.printSchema();
        usersDF.foreach(row => println(row))
        spark.close();
    }
}
